tools/distpartitioning/gloo_wrapper.py

```diff
@@ -59,6 +59,74 @@ def alltoall_cpu(rank, world_size, output_tensor_list, input_tensor_list):
     for i in range(world_size):
         dist.scatter(output_tensor_list[i], input_tensor_list if i == rank else [], src=i)
 
+def alltoallv_cpu_data(rank, world_size, input_tensor_list, val_dtype):
+    """
+    Wrapper function to providing the alltoallv functionality by using underlying alltoall
+    messaging primitive
+
+    Parameters:
+    -----------
+    rank : int
+        The rank of current worker
+    world_size : int
+        The size of the entire
+    input_tensor_list : List of tensor
+        The tensors to exchange
+    val_dtype : torch dtype
+        torch dtype which is used to create buffer for receiving messages
+
+    Returns:
+    --------
+    list : 
+        list of tensors received from other processes during alltoall message
+
+    """
+    sizes = [list(x.size()) for x in input_tensor_list]
+    for idx in range(1,len(sizes)):
+        assert len(sizes[idx-1]) == len(sizes[idx])
+
+    #decide how much to pad. 
+    #always use the first-dimension for padding. 
+    ll = [ x[0] for x in sizes ]
```

Comment:
if the input tensors are matrices, does the function still work? if the input tensors are 3D tensors, does the function still work?
if you want to make it general, you should flatten the input tensor to 1D vector and pad them. otherwise, you should make sure the input tensors are 1D vectors.

Reply:
or do you assume the input tensor may only differ in the first dimension. if so, you should check it explicitly.

Reply:
This function assumes that the tensors in the input_tensor_list can be of random dimensions. They ALL can be 1-D, 2-D or n-D. But all the tensors should be of the same size. Added 'assert' to ensure that all the input tensors are of same dimensions.

This function also assumes that only the FIRST dimension of the input tensors can differ and padding is needed. Added assert to check this condition as well.